pkg/migrate/marketplacemigration.go

```diff
@@ -0,0 +1,424 @@
+				Protocol: endpointDef.Protocol,
+				Routing: mv1a.ApiServiceInstanceSpecRouting{
+					BasePath: endpointDef.BasePath,
+				},
+			}
+			endPoints = append(endPoints, ep)
+		}
+	} else {
+		m.logger.Debug("Processing API service instance with no endpoint")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return endPoints, nil
+}
+
+func (m *MarketplaceMigration) handleSvcInstance(
+	svcInstance *v1.ResourceInstance, revision *v1.ResourceInstance,
+) error {
+	logger := m.logger.
+		WithField("instance-name", svcInstance.Name).
+		WithField("revision-name", revision.Name)
+
+	apiSvcInst := mv1a.NewAPIServiceInstance(svcInstance.Name, svcInstance.Metadata.Scope.Name)
+	apiSvcInst.FromInstance(svcInstance)
+
+	specProcessor, err := m.getSpecParser(revision)
+	if err != nil {
+		return err
+	}
+
+	endPoints, err := specProcessor.GetEndpoints()
+	instanceSpecEndPoints, err := m.createInstanceEndpoint(endPoints)
+	if err != nil {
+		return err
+	}
+
+	var i interface{} = specProcessor
+
+	if processor, ok := i.(apic.OasSpecProcessor); ok {
+		ardRIName := apiSvcInst.Spec.AccessRequestDefinition
+		credentialRequestPolicies := apiSvcInst.Spec.CredentialRequestDefinitions
+
+		processor.ParseAuthInfo()
+
+		// get the auth policy from the spec
+		authPolicies := processor.GetAuthPolicies()
+
+		// get the apikey info
+		apiKeyInfo := processor.GetAPIKeyInfo()
+		if len(apiKeyInfo) > 0 {
+			logger.Debug("instance has a spec definition type of apiKey")
+			ardRIName = "api-key"
+		}
+
+		// get oauth scopes
+		oauthScopes := processor.GetOAuthScopes()
+
+		var updateRequestDefinition = false
+
+		// Check if ARD exists
+		if apiSvcInst.Spec.AccessRequestDefinition == "" && len(oauthScopes) > 0 {
+			// Only migrate resource with oauth scopes. Spec with type apiKey will be handled on startup
+			logger.Debug("instance has a spec definition type of oauth")
+			ardRIName, err = m.processAccessRequestDefinition(oauthScopes)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Check if CRD exists
+		credentialRequestPolicies, err = m.getCredentialRequestPolicies(authPolicies)
+		if err != nil {
+			return err
+		}
+
+		// Find only the known CRDs
+		credentialRequestDefinitions := m.checkCredentialRequestDefinitions(credentialRequestPolicies)
+		if len(credentialRequestDefinitions) > 0 && !m.sortCompare(apiSvcInst.Spec.CredentialRequestDefinitions, credentialRequestDefinitions) {
+			log.Debugf("adding the following credential request definitions %s,", credentialRequestDefinitions)
+			updateRequestDefinition = true
+		}
+
+		existingARD, _ := m.cache.GetAccessRequestDefinitionByName(ardRIName)
+		if existingARD != nil && apiSvcInst.Spec.AccessRequestDefinition == "" {
+			logger.Debugf("adding the following access request definition %s", ardRIName)
+			updateRequestDefinition = true
+		}
+
+		if updateRequestDefinition {
+			inInterface := m.newInstanceSpec(instanceSpecEndPoints, revision.Name, ardRIName, credentialRequestDefinitions)
```

Comment:
apiSvcInst.Spec.Endpoint?

Reply:
Crap, yup, you're right.  Will fix

Reply:
using current spec from apiserviceinstance